BlueprintUI/Sources/Layout/GeometryReader.swift

```diff
@@ -48,8 +48,8 @@ public struct GeometryProxy {
     /// The size constraint of the element being laid out.
     public var constraint: SizeConstraint
 
-    /// Measure the given element, using this proxy's constraint and the current environment.
-    public func measure(element: Element) -> CGSize {
-        element.content.measure(in: constraint, environment: environment)
+    /// Measure the given element, using the provided constraint (or the proxy's constraint if not provided), and the current environment.
+    public func measure(in explicit: SizeConstraint? = nil, element: Element) -> CGSize {
```

Comment:
Another option is exposing the `environment`, so people can just perform their own measurements.

Reply:
Could/should we just update the `constraint` before measuring (in the consuming code) instead of making these changes?

Reply:
We could; but that's really weird (IMO) – the geometry reader's proxy is meant to represent the size of the reader itself. I'd rather not encourage folks to mutate it.

Reply:
Yeah, agreed about that feeling odd. In fact, though I’m generally a fan of struct members being `var`s across the board, I’d be in favor of `constraint` being a `let` (or `internal(set)`) here to make that distinction clearer.

Reply:
On the original topic, I prefer the approach in the PR currently over exposing `environment` here. Keeps the focus clear and is a little easier to discover IMO.